setup.py

```diff
@@ -56,7 +43,10 @@ def get_ucp_version():
 cmdclass = versioneer.get_cmdclass(cmdclass)
 
 setup(
+    name="ucx-py" + os.getenv("RAPIDS_PY_WHEEL_CUDA_SUFFIX", default=""),
```

Comment:
Am curious if this works given it is also in `pyproject.toml`. If so, great!

FWIW had tried adding `name` to the [`dynamic` attribute list]( https://github.com/rapidsai/ucx-py/blob/branch-0.29/pyproject.toml#L26-L28 ), which didn't work. AIUI this lines up with [the `dynamic` specification in PEP 621]( https://peps.python.org/pep-0621/#dynamic ):

> A build back-end MUST raise an error if the metadata specifies the `name` in `dynamic`.

Reply:
Should add if this doesn't work we may want to consider:

* Templating `pyproject.toml`
* Having multiple `pyproject.toml`s and using one based on CUDA version
* Some other solution that doesn't depend on changing `name` (for example using [local version identifiers]( https://peps.python.org/pep-0440/#local-version-identifiers ), altering other metadata, including a metapackage dependency, etc.)

Reply:
Ah good point I forgot about that. I do not think that the package name is being modified by the suffix at present. I'm waiting to get wheels built to verify everything though. 

I would be very happy if we could come up with a scheme that did not require this environment variable-based approach. I don't know how well anything plays with multiple or templated pyproject.toml files, though, and my understanding of local version identifiers is that they are meant to denote versions that are ABI compatible. I am not familiar with how else this might be done.

Reply:
I will say that the expedient solution would be to simply remove the name from pyproject.toml altogether. Certainly not my favorite solution, but I would be OK with it temporarily to get this release done while we search for a longer-term solution. We definitely need one across all of RAPIDS.

Reply:
I've temporarily removed the name from pyproject.toml because I want to get an up-to-date ucx build for testing raft wheels, but then I can revert until we can come up with a consensus on the best solution for this problem.

Reply:
It looks like that isn't valid either, the name must exist in that section. I'll undo this change and modify the downstream library's dependencies for now.

Reply:
Possibly silly question, but could we just hardcode the value? Know we are looking to handle other versions in the future, but right now it is just one value correct? Maybe we can skip customizing it until we need that

Reply:
We could. Will it affect anything with the conda build if the pyproject.toml name does not match the meta.yaml name?

Reply:
My guess is probably not, but let's try it and see 🙂

Reply:
Trying it now!

Reply:
OK it seems like it worked fine. When I look at the build logs ([here's one example](https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/rapidsai%2Fgpuci%2Fucx-py%2Fprb%2Fucx-py-cpu-build-arm64/detail/ucx-py-cpu-build-arm64/552/pipeline)) I see a few references to the `cu11` suffix, mostly coming from metadata files (egg-info, PKG-INFO, etc). @jakirkham does any of that look concerning to you in terms of how it will get packaged up by conda, or do you think the hardcoded solution should be fine?

Reply:
Yeah I don't think it should be an issue

Maybe there's some value in having this metadata match the wheel somewhat as that might protect against `pip` clobbering the Conda package. Could be worth testing this though

Reply:
@jakirkham in order to avoid this problem entirely, I've changed the name back in pyproject.toml and I've modified the wheels workflow to modify the pyproject.toml file at build time. I think this approach is cleaner for this release and avoids these concerns. For 23.02 I'm going to try to figure out the appropriate long-term solution, but I think this is suitable for now.

Reply:
[I've opened a discussion on the Python forums](https://discuss.python.org/t/dynamic-project-names-and-pep-621/21359) to continue this further. I'd like to see what solutions have been considered for this in the past before we try and come up with one of our own to apply across RAPIDS. The one that I've implemented in ucx-py seems fine for now.